apps/condo/domains/billing/utils/receiptQRCodeUtils.spec.js

```diff
@@ -42,31 +51,77 @@ describe('receiptQRCodeUtils', () => {
-            Field2: 'world',
-            foo: 'bar baz',
+    describe('RU QR-codes decoded correctly', () => {
+        const cases = [
+            [1, 'ÀÒÌÎÑÔÅÐÀ', 'АТМОСФЕРА'], // Got from real QR-code
+            // Got from conversion (https://convertcyrillic.com/)
+            // curl 'https://convertcyrillic.com/api/conversionservice' --data-raw '={"sourceEncoding":"Russian.Unicode","targetEncoding":"Russian.KOI8","text":"Привет"}'
+            [1, 'Ïðèâåò', 'Привет'],
+            [2, 'Привет', 'Привет'],
+            // curl 'https://convertcyrillic.com/api/conversionservice' --data-raw '={"sourceEncoding":"Russian.Unicode","targetEncoding":"Russian.CP1251","text":"Привет"}'
+            [3, 'ðÒÉ×ÅÔ', 'Привет'],
+        ]
+
+        test.each(cases)('%p %p -> %p', (tag, dataFromQRCode, expectedData) => {
+            const qrStr = `ST0001${tag}|field1=Hello|Field2=world|foo=${dataFromQRCode}`
+            const parsed = parseRUReceiptQRCode(qrStr)
+            expect(parsed).toEqual({
+                field1: 'Hello',
+                Field2: 'world',
+                foo: expectedData,
+            })
         })
     })
 
     test('must throw an error on invalid QR-code', async () => {
         await catchErrorFrom(
             async () => {
-                parseReceiptQRCode('ST0012|field1=Hello|Field2=world')
+                parseRUReceiptQRCode('ST0012|field1=Hello|Field2=world')
             },
             (err) => {
                 expect(err).toEqual(expect.objectContaining({ message: 'Invalid QR code' }))
             },
         )
     })
 
+    test('can get field case-insensitively', () => {
+        const qrStr = 'ST00012|field1=Hello|Field2=world|someField=!!'
+        const parsed = parseRUReceiptQRCode(qrStr)
+        expect(parsed).toEqual({
+            field1: 'Hello',
+            Field2: 'world',
+            someField: '!!',
+        })
+
+        expect(getQRCodeField(parsed, 'Field1')).toBe('Hello')
+        expect(getQRCodeField(parsed, 'SomeField')).toBe('!!')
+    })
+
+    test('Fields returned as requested', () => {
+        const qrStr = 'ST00012|field1=Hello|Field2=world|someField=!!|paymPeriod=01.2024'
+        const parsed = parseRUReceiptQRCode(qrStr)
+        const fields = getQRCodeFields(parsed, ['field1', 'FIELD2', 'somefield', 'PaymPeriod'])
+
+        expect(fields).toEqual({
+            field1: 'Hello',
+            FIELD2: 'world',
+            somefield: '!!',
+            PaymPeriod: '01.2024',
+        })
+    })
+
     test('check for required fields', () => {
-        const parsed = parseReceiptQRCode('ST00012|field1=Hello|Field2=world|foo=bar baz')
+        const parsed = parseRUReceiptQRCode('ST00012|field1=Hello|Field2=world|foo=bar baz')
+        const missedFields = getQRCodeMissedFields(parsed)
+
+        expect(missedFields).toEqual(['BIC', 'PayerAddress', 'Sum', 'PersAcc', 'PayeeINN', 'PersonalAcc'])
+    })
+
+    test('check for required fields except one', () => {
```

Comment:
what does this test check? it seems to be no different from the previous one

Reply:
Fixed